address.go

```diff
@@ -149,8 +161,9 @@ func DecodeAddress(addr string, defaultNet *chaincfg.Params) (Address, error) {
 			}
 
 			// We currently only support P2WPKH and P2WSH, which is
-			// witness version 0.
-			if witnessVer != 0 {
+			// witness version 0 and P2TR which is witness version
```

Comment:
We want to also have the `decodeSegWitAddress` method return the `bech32` version returned as well. This way we can reject decoding an address that's segwit v1, but that doesn't use the `bech32m` encoding.

Reply:
To accomplish this, we need to change the method to instead use `DecodeGeneric` so it can obtain the `bech32` version used when encoding.